component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1658,4 +1669,114 @@ private long getUnlockTimeInMilliSeconds(AuthenticatedUser authenticatedUser) th
+        eventProperties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, context.getTenantDomain());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN, authenticatedUser
+                .getUserStoreDomain());
+        eventProperties.put(IdentityEventConstants.EventProperty.APPLICATION_NAME, context.getServiceProviderName());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_AGENT, request.getHeader(
+                SMSOTPConstants.USER_AGENT));
+
+        if (request.getHeader(SMSOTPConstants.X_FORWARDED_FOR) != null) {
+            eventProperties.put(IdentityEventConstants.EventProperty.CLIENT_IP, request.getHeader(
+                    SMSOTPConstants.X_FORWARDED_FOR).split(",")[0]);
+        } else {
+            eventProperties.put(IdentityEventConstants.EventProperty.CLIENT_IP, request.getRemoteAddr());
+        }
+
+        eventProperties.put(IdentityEventConstants.EventProperty.GENERATED_OTP, context.getProperty(
+                SMSOTPConstants.OTP_TOKEN));
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_INPUT_OTP, request.getParameter(
+                SMSOTPConstants.CODE));
+        eventProperties.put(IdentityEventConstants.EventProperty.OTP_USED_TIME, System.currentTimeMillis());
+
+        long otpGeneratedTime = (long) context.getProperty(SMSOTPConstants.SENT_OTP_TOKEN_TIME);
+        eventProperties.put(IdentityEventConstants.EventProperty.OTP_GENERATED_TIME,
+                otpGeneratedTime);
+
+        String otpExpiryTime = SMSOTPUtils.getTokenExpiryTime(context);
+        if (otpExpiryTime != null) {
+            long expiryTime = otpGeneratedTime + Long.parseLong(otpExpiryTime);
+            eventProperties.put(IdentityEventConstants.EventProperty.OTP_EXPIRY_TIME, expiryTime);
+        }
+
+        String status;
+        if (context.getProperty(SMSOTPConstants.TOKEN_EXPIRED) != null && context.getProperty(
+                SMSOTPConstants.TOKEN_EXPIRED) == SMSOTPConstants.TOKEN_EXPIRED_VALUE) {
+            status = SMSOTPConstants.STATUS_OTP_EXPIRED;
+        } else if (context.getProperty(
+                SMSOTPConstants.CODE_MISMATCH) != null && (boolean) context.getProperty(
+                SMSOTPConstants.CODE_MISMATCH)) {
+            status = SMSOTPConstants.STATUS_CODE_MISMATCH;
+        } else {
+            status = SMSOTPConstants.STATUS_SUCCESS;
+        }
+
+        eventProperties.put(IdentityEventConstants.EventProperty.OTP_STATUS, status);
+        Event postOtpValidateEvent = new Event(IdentityEventConstants.Event.POST_VALIDATE_SMS_OTP, eventProperties);
+
+        try {
+            SMSOTPServiceDataHolder.getInstance().getIdentityEventService().handleEvent(postOtpValidateEvent);
+        } catch (IdentityEventException e) {
+            String errorMsg = "An error occurred while triggering the event. " + e.getMessage();
```

Comment:
Shall we add a more descriptive error message, as this exception occurred in the validation flow

Reply:
Fixed in 7d65437416ff6a2f9c995097fb0958fc95a58cf3